ludwig/features/image_feature.py

```diff
@@ -90,6 +92,31 @@
 ###
 # Image specific augmentation operations
 ###
+@register_augmentation_op(name="auto_augmentation", features=IMAGE)
+class AutoAugment(torch.nn.Module):
+    def __init__(self, config: AutoAugmentationConfig):
+        super().__init__()
+        self.auto_augmentation_method = config.method
+        self.augmentation_method = self.get_augmentation_method()
+
+    def get_augmentation_method(self):
+        if self.auto_augmentation_method == "trivial_augment":
+            return transforms.TrivialAugmentWide()
+        elif self.auto_augmentation_method == "auto_augment":
+            return transforms.AutoAugment()
+        elif self.auto_augmentation_method == "rand_augment":
+            return transforms.RandAugment()
+        else:
+            raise ValueError(f"Unsupported auto-augmentation method: {self.auto_augmentation_method}")
+
+    def forward(self, imgs):
+        method = self.augmentation_method
+        for i in range(imgs.size(0)):
```

Comment:
Is it possible to vectorize the augmentation operation so that it operates on the `imgs` tensor all at once?

Reply:
Hi @justinxzhao. I just pushed the changes that you described in the PR review. Thanks for helping get this feature out!